core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelFile.java

```diff
@@ -233,7 +241,10 @@ public String updateGlobalDefinition(String id, GlobalDefinitionCamelModelElemen
 		final Node parentNode = cme.getXmlNode().getParentNode();
 		final Element documentElement = getDocument().getDocumentElement();
 		if (parentNode == null || !parentNode.isEqualNode(documentElement)) {
-			documentElement.replaceChild(cme.getXmlNode(), oldDef);
+			// to avoid the occasional org.w3c.dom.DOMException: WRONG_DOCUMENT_ERR: 
+			// A node is used in a different document than the one that created it.
```

Comment:
This seems to have been resolved when we fixed the other model issues with things not being updated properly in memory.